### ISSUE TITLE
Problem with accounts-password (parties example)

### ISSUE THREAD

**Issue by sdarnell** [sdarnell/meteor-win#23]
Create the parties example on 0.7.0.1, fire it up, and then try to register an account.

On first attempt the client gets an internal server error with the following on the console:

```
Exception while invoking method 'createUser' TypeError: Object [object Object] has no method '_setLoginToken'
at Meteor.methods.createUser (packages/accounts-password/password_server.js:518)
at packages/check/match.js:77
at _.extend.withValue (packages/meteor/dynamics_nodejs.js:35)
at Object.Match._failIfArgumentsAreNotAllChecked (packages/check/match.js:76)
at maybeAuditArgumentChecks (packages/livedata/livedata_server.js:1346)
at packages/livedata/livedata_server.js:569
at _.extend.withValue (packages/meteor/dynamics_nodejs.js:35)
at packages/livedata/livedata_server.js:568
at _.extend.withValue (packages/meteor/dynamics_nodejs.js:35)
at _.extend.protocol_handlers.method (packages/livedata/livedata_server.js:567)
```

It seems to create some mongo record though because if you try again with the same email you get a different error:

```
Exception while invoking method 'beginPasswordExchange' TypeError: Cannot set property 'srpChallenge' of undefined
    at Meteor.methods.beginPasswordExchange (packages/accounts-password.js:163:34)
    at packages/check/match.js:77
    at _.extend.withValue (packages/meteor/dynamics_nodejs.js:35)
    at Object.Match._failIfArgumentsAreNotAllChecked (packages/check/match.js:76)
    at maybeAuditArgumentChecks (packages/livedata/livedata_server.js:1346)
    at packages/livedata/livedata_server.js:569
    at _.extend.withValue (packages/meteor/dynamics_nodejs.js:35)
    at packages/livedata/livedata_server.js:568
    at _.extend.withValue (packages/meteor/dynamics_nodejs.js:35)
    at _.extend.protocol_handlers.method (packages/livedata/livedata_server.js:567)
```

The original report comes from https://groups.google.com/forum/#!topic/meteor-talk/SyDPSIcV-og
and notes that it is fine in 0.6.6.3.  (Christian Knappke).

**Comment by sdarnell** [comment]
David Glasser followed up on the group with:
That error would come from a line of code that isn't actually in 
0.7.0.1.  ie, the 0.6.6.3 version has 

```
  this._sessionData.srpChallenge = 
    { userId: user._id, M: srp.M, HAMK: srp.HAMK }; 
```

The 0.7.0.1 version has (using a new API) 

```
  Accounts._setAccountData(this.connection.id, 'srpChallenge', 
    { userId: user._id, M: srp.M, HAMK: srp.HAMK } 
```

It sounds like you are somehow combining a 0.6.6.3 accounts-password 
package with the 0.7.0.1 accounts-base package.  This does sound like 
a problem with the Windows fork, though I dunno, 
https://github.com/sdarnell/meteor/blob/win0.7.0.1/packages/accounts-password/password_server.js 
shows the new code.

**Comment by sdarnell** [comment]
Looking at the packages, the dist directory has the right code but the installed version doesn't. Interestingly the hash for the package is the same (f5c4d6f200) which sounds wrong. So it is probably something that went wrong during the release process.

**Comment by sdarnell** [comment]
There seems to have been some mistake during the release process which meant that the package ended up with the same hash, and I'm pretty sure that is the root cause of the problem.

I'll look at getting a fix out, but here is a work-around. The 0.7.0.1 bootstrap package includes the right file so a fresh install should work (I've tested it and it does for me).

The simplest way of doing this is to move your the '%LOCALAPPDATA%.meteor' directory sideways to `.meteor.sav` (or remove it if you don't need 0.6.6.3), then re-run the LaunchMeteor.exe installer.

**Comment by k5e** [comment]
Thanks Stephen, that helped.

**Comment by zimt28** [comment]
Same error here

**Comment by sdarnell** [comment]
The problem was due to the way the packages were build and released, meaning that their hashes weren't calculated correctly. Fixed in commit c98d159a988cc48c1325ac7c5ad18508c43fe5b8
